raster-foundry/raster-foundry
# Add re-designed permissions modal

## Overview

This PR implements the re-designed permissions modal.

More specifically it:
- adds suggestion display and selection to the search component
- adds some resolves to various route trees to make the user's platform available in more places
- removes the old permissions modal
- adds the new one!

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] [Swagger specification](https://github.com/raster-foundry/raster-foundry-api-spec) updated~
~- [ ] Symlinks from new migrations present or corrected for any new migrations~
~- [ ] Any content changes are properly templated using `BUILDCONFIG.APP_NAME`~
~- [ ] Any new SQL strings have tests~

### Demo

![image](https://user-images.githubusercontent.com/2442245/43154167-626c8136-8f41-11e8-800e-a2c0c53e5171.png)

![image](https://user-images.githubusercontent.com/2442245/43154188-76898786-8f41-11e8-99b1-80d40dd41840.png)

## Testing Instructions

- try adding different permissions to different projects and scenes and check that it is sending the appropriate requests
- try to break the auto-suggest (not _too_ hard, but in reasonable way)
- make sure you can't add duplicates
- make sure you can go from having permissions on an object to having none and back


Closes https://github.com/azavea/raster-foundry-platform/issues/357
